sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/models/Headers.java

```diff
@@ -293,11 +293,11 @@ private String getValueInternal(String formattedName) {
      *
      * @return The values of the {@link Header}, or {@code null} if the {@link Header} isn't found.
      */
-    public String[] getValues(HeaderName name) {
+    public List<String> getValues(HeaderName name) {
```

Comment:
I think we should remove the APIs that are just conveniences around `Header`, like this. Push the usage to `Header getValue(HeaderName)` and then calling `Header.getValues()`.

This way we don't need to update `Headers` if we make API changes to `Header`.

Reply:
Checking for nullability every time you call can get verbose, this is from one of our tests, for example:
```java
Headers headers = request.getHeaders();

assertEquals("baz", headers.get(X_MS_FOO).getValue());
```
Which would end up looking like this:
```java
Headers headers = request.getHeaders();
Header xMsFoo = headers.get(X_MS_FOO);

assertEquals("baz", xMsFoo == null ? null : xMsFoo.getValue());
```

The same goes for `getValues()`. Are we sure we want to do that?